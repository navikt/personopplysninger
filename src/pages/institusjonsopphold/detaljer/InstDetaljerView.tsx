import { FormattedMessage } from 'react-intl';
import { BodyShort, Heading, HelpText } from '@navikt/ds-react';
import dayjs from 'dayjs';
import ListElement from '@/components/listelement/ListElement';
import { formatOrgnr, RADIX_DECIMAL } from '@/utils/formattering';
import { InstInnslag } from '@/types/inst';
import { Liste } from '@/components/listelement/Liste';

const InstDetaljerView = (props: { innslag: InstInnslag }) => {
    const { innslag } = props;
    const startdato = dayjs(innslag.startdato).format('DD.MM.YYYY');
    const faktiskSluttdato = innslag.faktiskSluttdato ? dayjs(innslag.faktiskSluttdato).format('DD.MM.YYYY') : '';

    return (
        <div>
            <div className="detaljer__tittel">
                <Heading level="2" size="small">
                    {innslag.institusjonsnavn}
                </Heading>
                {innslag.organisasjonsnummer && (
                    <BodyShort>
                        <FormattedMessage
                            id="side.organisasjonsnummer"
                            values={{
                                orgnr: formatOrgnr(parseInt(innslag.organisasjonsnummer, RADIX_DECIMAL).toString()),
                            }}
                        />
                    </BodyShort>
                )}
            </div>
            <hr className="box__linje-bred" />
            <div className="box">
                <div className="box__content">
                    <Liste>
                        <ListElement titleId={'inst.institusjonstype'} content={innslag.institusjonstype} />
                        <ListElement
                            titleId={'inst.periode'}
                            content={
                                <div className={'inst__periode'}>
                                    {`${startdato} - ${faktiskSluttdato}`}
                                    {innslag.fiktivSluttdato && (
                                        <HelpText>
                                            <FormattedMessage id={'inst.fiktivSluttdato'} />
                                        </HelpText>
                                    )}
                                </div>
                            }
                        />
                    </Liste>
                </div>
            </div>
        </div>
    );
};

export default InstDetaljerView;
