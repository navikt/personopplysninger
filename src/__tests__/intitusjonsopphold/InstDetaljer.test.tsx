import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import nbMessages from '@/text/nb';
import instInfo from '@/clients/apiMock/app/fetch/inst-info.json';
import InstDetaljerView from '@/pages/institusjonsopphold/detaljer/InstDetaljerView';

vi.mock('react-router-dom', () => ({
    Link: () => <div />,
    useLocation: () => ({
        pathname: 'pathname',
    }),
}));

describe('InstDetaljerView', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <IntlProvider locale={'nb'} messages={nbMessages}>
                <InstDetaljerView innslag={instInfo[0]} />
            </IntlProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
