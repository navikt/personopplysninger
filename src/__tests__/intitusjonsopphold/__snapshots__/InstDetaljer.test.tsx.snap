// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InstDetaljerView > renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="detaljer__tittel"
    >
      <h2
        class="navds-heading navds-heading--small"
      >
        VIK FENGSEL
      </h2>
      <p
        class="navds-body-short navds-body-short--medium"
      >
        Organisasjonsnummer 974 736 942
      </p>
    </div>
    <hr
      class="box__linje-bred"
    />
    <div
      class="box"
    >
      <div
        class="box__content"
      >
        <dl
          class="list"
        >
          <div
            class="list-item"
          >
            <dt
              class="list-element-heading"
            >
              <span
                class="navds-label"
              >
                Institusjonstype
              </span>
            </dt>
            <dd>
              FO
            </dd>
          </div>
          <div
            class="list-item"
          >
            <dt
              class="list-element-heading"
            >
              <span
                class="navds-label"
              >
                Periode
              </span>
            </dt>
            <dd>
              <div
                class="inst__periode"
              >
                13.06.2017 - 15.09.2017
              </div>
            </dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
